File: HakerRank/data_structures/linked_lists/insert_a_node_at_the_tail_of_a_linked_list.py
# author: Daniel Lozano
# source: HackerRank ( https://www.hackerrank.com )
# problem name: Data Structures: Linked Lists: Insert a Node at the Tail of a Linked List
# problem url: https://www.hackerrank.com/challenges/insert-a-node-at-the-tail-of-a-linked-list/problem
# date: 9/1/2017

 class Node(object):
 
   def __init__(self, data=None, next_node=None):
       self.data = data
       self.next = next_node
       
def Insert(head, data):
    instance = Node(data)
    current = head
    if head is None:
        return instance
    else:
        while current.next is not None:
            current = current.next
        current.next = instance
        return head

File: HakerRank/data_structures/linked_lists/insert_a_node_at_the_tail_of_a_linked_list.cs
/*
author: Daniel Lozano
source: HackerRank ( https://www.hackerrank.com )
problem name: Data Structures: Linked Lists: Insert a Node at the Tail of a Linked List
problem url: https://www.hackerrank.com/challenges/insert-a-node-at-the-tail-of-a-linked-list/problem
*/

using System;

namespace HackerRank
{
    public class Program
    {
        public class Node
        {
            public int Data;
            public Node Next;
        }

        public static Node Insert(Node head, int x)
        {
            Node node = new Node() { Data = x };
            if (head == null)
                return node;
            Node current = head;
            while (current.Next != null)
                current = current.Next;
            current.Next = node;
            return head;
        }

        public static void Main()
        {
           
        }
    }
}